client/src/components/subscription/subscriptionDetail.js

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTimes } from '@fortawesome/free-solid-svg-icons';
+import { parseWeeklyDates, parseMonthlyDates } from '../../utils/frontendDateUtils';
+import DetailCalendar from '../date/detailCalendar';
+
+const SubscriptionDetail = ({ setActiveSubscription, details, setEditingSubscription, handleDelete }) => {
+  const { name, nickname, dueDate, reminderDays, amount, subscriptionUuid } = details;
+
+  const [repeatString, setRepeatString] = useState('');
+
+  useEffect(() => {
+    let parsedRepeatString;
+
+    if (dueDate.frequency === 'weekly') {
+      parsedRepeatString = (dueDate.occurrence > 1)
+        ? `Every ${dueDate.occurrence} weeks on ${parseWeeklyDates(dueDate.dates)}.`
+        : `Every week on ${parseWeeklyDates(dueDate.dates)}.`;
+    } else if (dueDate.frequency === 'monthly') {
+      parsedRepeatString = (dueDate.occurrence > 1)
+        ? `Every ${dueDate.occurrence} months on ${parseMonthlyDates(dueDate.dates)}.`
+        : `Every month on ${parseMonthlyDates(dueDate.dates)}.`;
+    } else if (dueDate.frequency === 'daily') {
+      parsedRepeatString = (dueDate.occurrence > 1)
+        ? `Every ${dueDate.occurrence} days.`
+        : 'Every day.';
+    } else {
+      parsedRepeatString = 'Every year.';
+    }
+
+    setRepeatString(`${new Date(dueDate.nextDueDate).toLocaleDateString()}, ${parsedRepeatString}`);
+  }, [name, dueDate]);
+
+  const handleEdit = useCallback((e) => {
+    e.stopPropagation();
+    setActiveSubscription(false);
+    setEditingSubscription(details);
+  }, [setActiveSubscription, setEditingSubscription, details]);
+
+  const clickDelete = useCallback((e) => {
+    e.stopPropagation();
+    setActiveSubscription(false);
+    handleDelete(subscriptionUuid);
+  }, [setActiveSubscription, handleDelete, subscriptionUuid]);
```

Comment:
we can handle this in the future, but I think having a confirmation modal (Are you sure you want to delete this?) would be good here

Reply:
Implemented in the task/addModal PR.